test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
Given `ProxyB.proxifiedObject = {foo: 1}` .
After 
```js
ProxyA.newValue = ProxyB.proxifiedObject
ProxyA.newValue.add = 1
```
will `ProxyB.proxifiedObject.add` be `1` or `undefined`?